syntax_body/java/syntax/pattern.vim

```diff
@@ -1,5 +1,6 @@
 "---------------------------------------------------------------------------------------------------
-sy match  javaOperator      '[()\.\[\]+\-\~!\*/%<>=&\^|?:,]'
+sy match  javaOperator      '\(\~\|!\|\*\|/\|%\| [<>]\|[<>] \|=\|&\|\^\||\|?\|:\)'
+sy match  javaDelimiter     '\((\|)\|\.\|\[\|\]\|+\-\|,\|[<>]\|;\)' contains=javaOperator
```

Comment:
The specification of `<>` on both `javaOperator` and `javaDelimiter` is so that generics will be highlighted as delimiters but greaterthan/lessthan will be highlighted as operators.